website/src/hooks/tasks/useGenericTaskAPI.tsx

```diff
@@ -26,10 +26,10 @@ export const useGenericTaskAPI = <TaskType,>(taskApiEndpoint: string) => {
   );
 
   useEffect(() => {
-    if (tasks.length === 0 && !isLoading && !error) {
+    if (data === undefined && !isLoading && !error) {
```

Comment:
Seems that change `undefined` broke fix #393. 
![demo](https://user-images.githubusercontent.com/30481105/211213374-965d0928-89e7-4ed3-8c59-3781cd5bb42c.png)

Reply:
oh, how to reproduce?

Reply:
should be fixed now, could you please check again?

Reply:
Tested. Fix is working!

Reply:
Hmm. It's working 90% of the time. But when rapidly navigating it seems broken.
![output3](https://user-images.githubusercontent.com/30481105/211214158-c6ddfb20-03b8-4a9f-8ef2-05c7b0564db3.gif)

Reply:
Maybe we should leave the check as it was? And wait for #551 to solve it?

Reply:
I reduced the dedupe time to half a second, which should be reasonable.

#551 might solve this entirely, I don't know if we should squeeze it before MVP.

Reply:
Let's get the random task change done then we can switch to `getServerSideProps`.  That can be rolled out anytime.